src/fstext/deterministic-fst.h

```diff
@@ -84,10 +84,29 @@ class DeterministicOnDemandFst {
 
   /// Note: ilabel must not be epsilon.
   virtual bool GetArc(StateId s, Label ilabel, Arc *oarc) = 0;
-
   virtual ~DeterministicOnDemandFst() { }
 };
 
+template<class Arc>
+class DeterministicOnDemandFstParallel {
+ public:
+  typedef typename Arc::StateId StateId;
+  typedef typename Arc::Weight Weight;
+  typedef typename Arc::Label Label;
+
+  virtual StateId Start() = 0;
+
+  // dgao
```

Comment:
this comment shouldn't be here, and various aspects of this aren't right, both from a Google style guid point of view and a documentation point of view. I'm not going to get into details as this doesn't seem review-ready at this point.f

Reply:
Oh right. We will spend some time carefully going over all the code. I will send you an explicit "ready-to-review" message when I feel it's ready.